test/unit/mdc-chips/mdc-chip.foundation.test.js

```diff
@@ -99,6 +101,80 @@ test('on click, emit custom event', () => {
   td.verify(mockAdapter.notifyInteraction());
 });
 
+test(`on leading icon opacity transition end, add ${cssClasses.HIDDEN_LEADING_ICON}` +
+  'class to leading icon if chip is selected', () => {
+  const {foundation, mockAdapter} = setupTest();
+  const handlers = captureHandlers(mockAdapter, 'registerEventHandler');
+  const mockEvt = {
+    type: 'transitionend',
+    target: {},
+    propertyName: 'opacity',
+  };
+  td.when(mockAdapter.eventTargetHasClass(mockEvt.target, cssClasses.LEADING_ICON)).thenReturn(true);
+  td.when(mockAdapter.hasClass(cssClasses.SELECTED)).thenReturn(true);
+
+  foundation.init();
+  handlers.transitionend(mockEvt);
+
+  td.verify(mockAdapter.addClassToLeadingIcon(cssClasses.HIDDEN_LEADING_ICON));
+});
+
+test('on leading icon opacity transition end, do nothing if chip is not selected', () => {
+  const {foundation, mockAdapter} = setupTest();
+  const handlers = captureHandlers(mockAdapter, 'registerEventHandler');
+  const mockEvt = {
+    type: 'transitionend',
+    target: {},
+    propertyName: 'opacity',
+  };
+  td.when(mockAdapter.eventTargetHasClass(mockEvt.target, cssClasses.LEADING_ICON)).thenReturn(true);
+  td.when(mockAdapter.hasClass(cssClasses.SELECTED)).thenReturn(false);
+
+  foundation.init();
+  handlers.transitionend(mockEvt);
+
+  td.verify(mockAdapter.addClassToLeadingIcon(cssClasses.HIDDEN_LEADING_ICON), {times: 0});
+});
+
+test(`on checkmark opacity transition end, remove ${cssClasses.HIDDEN_LEADING_ICON}` +
+  'class from leading icon if chip is not selected', () => {
+  const {foundation, mockAdapter} = setupTest();
+  const handlers = captureHandlers(mockAdapter, 'registerEventHandler');
+  const mockEvt = {
+    type: 'transitionend',
+    target: {
+      classList: ['mdc-chip__checkmark'],
```

Comment:
Is this even necessary here and in the next test, when you're mocking the adapter calls? It wasn't necessary in the previous 2 tests...